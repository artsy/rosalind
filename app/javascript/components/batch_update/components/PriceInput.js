import React from 'react'
import TextInput from './TextInput'

const PriceInput = ({ minPrice, maxPrice, updateState }) => {
  const onSetMin = value => updateState('minPrice', value)
  const onSetMax = value => updateState('maxPrice', value)

  return (
    <div>
      {minPrice == null && (
        <TextInput
          className="minPriceInput"
          placeholder="Minimum Price"
          onEnter={onSetMin}
          numeric
        />
      )}
      {maxPrice == null && (
        <TextInput
          className="maxPriceInput"
          placeholder="Maximum Price"
          onEnter={onSetMax}
          numeric
        />
      )}
    </div>
  )
}

export default PriceInput
