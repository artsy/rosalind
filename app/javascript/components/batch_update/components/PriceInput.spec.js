import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import PriceInput from './PriceInput'
import { mount } from 'enzyme'

describe('PriceInput', () => {
  let component
  let updateState
  let wrapper
  let tree
  let minPriceInput
  let maxPriceInput
  const init = (options = {}) => {
    let { minPrice, maxPrice } = options
    updateState = jest.fn()
    component = (
      <PriceInput
        updateState={updateState}
        minPrice={minPrice}
        maxPrice={maxPrice}
      />
    )
    wrapper = mount(component)
    minPriceInput = wrapper.find('.minPriceInput input')
    maxPriceInput = wrapper.find('.maxPriceInput input')
    tree = renderer.create(component).toJSON()
  }
  it('renders correctly', () => {
    init()
    expect(tree).toMatchSnapshot()
  })
  it('correctly sets the min price', () => {
    init()
    minPriceInput.simulate('keyUp', { key: 'Enter', target: { value: '123' } })
    expect(updateState).toHaveBeenCalledWith('minPrice', 123)
  })
  it('correctly sets the max price', () => {
    init()
    maxPriceInput.simulate('keyUp', { key: 'Enter', target: { value: '123' } })
    expect(updateState).toHaveBeenCalledWith('maxPrice', 123)
  })
  it('ignores attempts to set a non-numeric value', () => {
    init()
    minPriceInput.simulate('keyUp', { key: 'Enter', target: { value: 'abc' } })
    maxPriceInput.simulate('keyUp', { key: 'Enter', target: { value: 'xyz' } })
    expect(updateState).not.toHaveBeenCalled()
  })
  it('both inputs exist by default', () => {
    init()
    expect(wrapper.exists('.minPriceInput')).toBe(true)
    expect(wrapper.exists('.maxPriceInput')).toBe(true)
  })
  it('does not display minPriceInput if minPrice is already set', () => {
    init({ minPrice: 1000 })
    expect(wrapper.exists('.minPriceInput')).toBeFalsy()
  })
  it('does not display maxPriceInput if maxPrice is already set', () => {
    init({ maxPrice: 1000 })
    expect(wrapper.exists('.maxPriceInput')).toBeFalsy()
  })
  it('does not display anything if both values are set', () => {
    init({ minPrice: 500, maxPrice: 1000 })
    expect(wrapper.exists('.minPriceInput')).toBeFalsy()
    expect(wrapper.exists('.maxPriceInput')).toBeFalsy()
  })
})
