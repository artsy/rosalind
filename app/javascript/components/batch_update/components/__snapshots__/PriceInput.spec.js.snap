// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInput renders correctly 1`] = `
.c0 input {
  width: 100%;
  margin: 1em 0 0 0;
  line-height: 2em;
  font-size: 1em;
  border: none;
  border-bottom: solid 1px #ddd;
  background: none;
}

.c0 input:focus {
  outline: none;
}

<div>
  <div
    className="c0 minPriceInput"
  >
    <input
      onKeyUp={[Function]}
      placeholder="Minimum Price"
      type="text"
    />
  </div>
  <div
    className="c0 maxPriceInput"
  >
    <input
      onKeyUp={[Function]}
      placeholder="Maximum Price"
      type="text"
    />
  </div>
</div>
`;
